lib/record.js

```diff
@@ -92,11 +92,20 @@ export default class Record {
     if (id === null || id === undefined) {
       id = uuid.v4();
     } else {
-      let [type, name] = Record.parseID(id);
-      if (type !== this._recordType) {
-        throw new Error('_id is not valid. RecordType mismatch.');
+      let tuple = null;
+      try {
+        tuple = Record.parseID(id);
+      } catch (e) {
+        // id is typeless
+        // default to recordType
+      }
+      if (tuple) {
+        let [type, name] = tuple;
+        if (type !== this._recordType) {
```

Comment:
Perhaps you also want to check if `_recordType` exists, and that it equals `this._recordType`.